src/cpp/core/r_util/RSourceIndex.cpp

```diff
@@ -366,6 +366,42 @@ void libraryCallIndexer(const RTokenCursor& cursor,
    }
 }
 
+void testThatCallIndexer(const RTokenCursor& cursor,
+                         const IndexStatus& status,
+                         bool isReadOnlyFile, 
+                         RSourceIndex* pIndex)
+{
+   if (!cursor.isType(RToken::ID))
+      return;
+   
+   if (!(cursor.contentEquals(L"test_that")))
+      return;
+   
+   RTokenCursor clone = cursor.clone();
+   if (!clone.moveToNextToken())
+      return;
+   
+   if (!clone.isType(RToken::LPAREN))
+      return;
+   
+   if (!clone.moveToNextToken())
```

Comment:
nit: use `moveToNextSignificantToken()` to handle cases like `test_that ( "desc"`; that is, cases where the user has included some extra whitespace.